warehouse/transformer/safe.go

```diff
@@ -0,0 +1,273 @@
+	}
+	if !intrOpts.skipReservedKeywordsEscaping && reservedkeywords.IsTableOrColumn(destType, name) {
+		name = "_" + name
+	}
+	if utils.IsDataLake(destType) {
+		return name
+	}
+	return misc.TruncateStr(name, 127)
+}
+
+func transformTableNameCached(tec *transformEventContext, key string) string {
+	var tableName string
+
+	cacheKey := transformTableNameCacheKey{
+		underscoreDivideNumbers: tec.destOpts.underscoreDivideNumbers,
+		useBlendoCasing:         tec.intrOpts.useBlendoCasing,
+		tableName:               key,
+	}
+	columnNameCached, ok := tec.cache.transformTableNameCache.Load(cacheKey)
+	if !ok {
+		tableName = transformTableName(tec.intrOpts, tec.destOpts, key)
+		tec.cache.transformTableNameCache.Store(cacheKey, tableName)
+	} else {
+		tableName = columnNameCached.(string)
+	}
+	return tableName
+}
+
+// transformTableName applies transformation to the input table name based on the destination type and configuration options.
+// If `useBlendoCasing` is enabled, it converts the table name to lowercase and trims spaces.
+// Otherwise, it applies a more general transformation using the `transformName` function.
+func transformTableName(intrOpts *intrOptions, destOpts *destOptions, tableName string) string {
+	if intrOpts.useBlendoCasing {
+		return strings.TrimSpace(strings.ToLower(tableName))
+	}
+
+	var snakeCaseFn func(s string) string
+	if destOpts.underscoreDivideNumbers {
+		snakeCaseFn = snakecase.ToSnakeCase
+	} else {
+		snakeCaseFn = snakecase.ToSnakeCaseWithNumbers
+	}
+
+	name := strings.Join(extractAlphanumericValues(tableName), "_")
+	if strings.HasPrefix(tableName, "_") {
+		name = reLeadingUnderscores.FindString(tableName) + snakeCaseFn(reLeadingUnderscores.ReplaceAllString(name, ""))
+	} else {
+		name = snakeCaseFn(name)
+	}
+	if startsWithDigit(name) {
+		name = "_" + name
+	}
+	return name
+}
+
+func transformColumnNameCached(tec *transformEventContext, key string) string {
+	var columnName string
+
+	cacheKey := transformColumnNameCacheKey{
+		destType:                tec.event.Metadata.DestinationType,
+		columnName:              key,
+		underscoreDivideNumbers: tec.destOpts.underscoreDivideNumbers,
+		useBlendoCasing:         tec.intrOpts.useBlendoCasing,
+	}
+	columnNameCached, ok := tec.cache.transformColumnNameCache.Load(cacheKey)
+	if !ok {
+		columnName = transformColumnName(tec.event.Metadata.DestinationType, tec.intrOpts, tec.destOpts, key)
+		tec.cache.transformColumnNameCache.Store(cacheKey, columnName)
+	} else {
+		columnName = columnNameCached.(string)
+	}
+	return columnName
+}
+
+// transformColumnName applies transformation to the input column name based on the destination type and configuration options.
+// If `useBlendoCasing` is enabled, it transforms the column name into Blendo casing.
+// Otherwise, it applies a more general transformation using the `transformName` function.
+func transformColumnName(destType string, intrOpts *intrOptions, destOpts *destOptions, columnName string) string {
```

Comment:
`transformColumnName` shares almost all the the code with the `transformTableName` function. Should we extract the common code ?

Reply:
It's too much of an abstraction to get this common code to a separate function.